host/fs/fs_linux.go

```diff
@@ -164,64 +164,34 @@ func (e *eventsListener) init() error {
 	case err.Error() == "function not implemented":
 		// Some arch (arm64) do not implement EpollCreate().
 		if e.epollFd, err = syscall.EpollCreate1(0); err != nil {
+			e.mu.Unlock()
 			return err
 		}
 	default:
+		e.mu.Unlock()
 		return err
 	}
 	e.r, e.w, err = os.Pipe()
-	e.fds = map[int32]chan<- time.Time{}
 	// Only need epollIN. epollPRI has no effect on pipes.
 	const flags = epollET | epollIN
 	if err := e.addFdInner(e.r.Fd(), flags); err != nil {
+		// This object will not be reusable at this point.
+		e.mu.Unlock()
 		return err
 	}
 	wakeUp := make(chan time.Time)
-	e.fds[int32(e.r.Fd())] = wakeUp
 	e.wakeUp = wakeUp
-	// The mutex is still held after this function exits, it's loop() that will
-	// release the mutex.
+	e.fds = map[int32]chan<- time.Time{}
+	e.fds[int32(e.r.Fd())] = wakeUp
+	// The mutexes are still held after this function exits, it's loop() that will
+	// release the mutexes.
 	//
 	// This forces loop() to be started before addFd() can be called by users.
 	e.wg.Add(1)
```

Comment:
nothing waits for this wait group.

Reply:
Thanks for catching this, removed the WaitGroup.